pkg/docker/config/config.go

```diff
@@ -183,7 +190,26 @@ func RemoveAllAuthentication(sys *types.SystemContext) error {
 			}
 			logrus.Debugf("error removing credentials from kernel keyring")
 		}
+		if len(auths.CredHelpers) != 0 {
+			for cred, registry := range auths.CredHelpers {
+				if err := deleteAuthFromCredHelper(cred, registry); err != nil {
+					// ingore the error if no credential stored for the registry
+					if credentials.IsCredentialsMissingServerURLMessage(err.Error()) {
+						continue
+					}
+					logrus.Debugf("error removing credentials for %s from credential helper %s: %q", registry, cred, err)
+					return false, err
+				}
+			}
+		}
+		if auths.CredsStore != "" {
+			if err := deleteAuthFromCredsStore(auths.CredsStore); err != nil {
+				logrus.Debugf("error removing credentials from credential helper %s: %q", auths.CredsStore, err)
+				return false, err
+			}
+		}
 		auths.CredHelpers = make(map[string]string)
+		auths.CredsStore = ""
```

Comment:
I’m not immediately sure deleting this makes sense — I’ll need to revisit previous discussions.